### PR TITLE
Improved definitions of Video.js 6+ for to implement custom Techs

### PR DESCRIPTION
**Pull request by cipacda**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/video.js/index.d.ts [hunk 1]
```diff
@@ -72,4 +74,137 @@ declare namespace videojs {
...
 		playbackRate(rate?: number): number;
 	}
+
+	interface Component {
+		options_: any;
+		el_: HTMLElement;
+		player_: any;
+		isReady_: boolean;
+
+    new(options: any, ready: () => void);
```
Comment (andy-ms) [inline types/video.js/index.d.ts]: Did you mean `class Component` and `constructor(...)`? This declares an interface for objects `x` s.t. you can call `new x(...)` and also `x.dispose()` etc.

File: types/video.js/index.d.ts [hunk 2]
```diff
@@ -72,4 +74,137 @@ declare namespace videojs {
...
+		setTimeout(fn: () => void, timeout?: number): string;
+    dispose(): void;
+    trigger(event: any): void;
+    triggerReady(): void;
+	}
+
+	type TechType = Tech;
```
Comment (andy-ms) [inline types/video.js/index.d.ts]: This shouldn't be necessary, just use `Tech`

File: types/video.js/tsconfig.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
             "es6",
             "dom"
         ],
-        "noImplicitAny": true,
+        "noImplicitAny": false,
```
Comment (andy-ms) [inline types/video.js/tsconfig.json]: Please keep this `true`

**Review by ghost — CHANGES_REQUESTED** [review]
The file currently uses tabs, please stick to that.

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has unaddressed comments and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**